### DISCUSSION TITLE
Cannot create PAT _with_ scope write:packages _without_ repo scope

### DISCUSSION THREAD

**Discussion by partiallyordered** [community/community#26673] [Actions]
<p>Per the documentation here:</p><aside class="onebox githubfolder">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">
      <a href="https://docs.github.com/en/packages/guides/pushing-and-pulling-docker-images" target="_blank" rel="noopener nofollow ugc">docs.github.com</a>
  </header>
  <article class="onebox-body">
    

<h3><a href="https://docs.github.com/en/packages/guides/pushing-and-pulling-docker-images" target="_blank" rel="noopener nofollow ugc">Pushing and pulling Docker images - GitHub Docs</a></h3>

<p><a href="https://docs.github.com/en/packages/guides/pushing-and-pulling-docker-images" target="_blank" rel="noopener nofollow ugc">//docs.github.com/en/packages/guides/pushing-and-pulling-docker-images</a></p>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<blockquote>
<p><strong>Note:</strong> If you select the <code>write:packages</code> scope, deselect the <code>repo</code> scope when creating the PAT.</p>
</blockquote>
<p>However, the UI does not appear to allow this; when write:packages is selected, I am unable to deselect the parent repo scope, or any child repo scope.</p>
<p>I am creating a PAT here: <a href="https://github.com/settings/tokens/new" rel="noopener nofollow ugc">https://github.com/settings/tokens/new</a></p>

**Comment by masip85** [comment]
<aside class="quote no-group" data-username="msk" data-post="1" data-topic="154300">
<div class="title">
<div class="quote-controls"></div>
 msk:</div>
<blockquote>
<p>he UI does not appear to allow this; when write:packages is selected, I am unable to deselect the parent repo scope, or any child repo scope.</p>
</blockquote>
</aside>
<p>Same issue creating my token</p>

**Comment by whitneyimura** [comment, marked as answer]
<p>:wave: Thanks for reporting this bug.  We have an issue to track and fix on our end.</p>
<p>In the meantime, you can create the PAT with the repo scope, and then edit it  to remove the repo scope.  I realize this isn’t ideal, but should be a workaround until the UI fix is done.</p>

**Comment by mykter** [comment]
<p>How does one edit an existing PAT’s scopes? There doesn’t seem to be an option to do anything except delete one that I can see.<br>
I was able to create a PAT without the repo scope by modifying the create token POST request with a proxy.</p>

**Comment by partiallyordered** [comment]
<p>The access tokens appear as a list, which I guess is where you can see the delete button. The access token description appears to me as blue, bold text. It is a link. Clicking the link will take you to a page where you can modify the scopes. (I agree this doesn’t seem especially intuitive when there is a button for delete and not for edit).</p>

**Comment by bixu** [comment]
<p>Even after creating the PAT, saving it, then attempting to edit the saved PAT’s scopes, it’s not possible to <em>keep</em> <code>write:packages</code> and <em>remove</em> <code>repo</code> scope :frowning:</p>

**Comment by PaulRudin** [comment]
<p>Yup; it appears that it’s no longer possible to remove the repo scope through the UI if you have the write packages scope (although I’m pretty sure you could do it a few weeks ago).</p>

**Comment by ziegfried** [comment]
<p>It still seems to work when modifying the form through the browser console and then saving it.</p>
<p>eg. by using</p>
<pre><code class="lang-auto">[...document.querySelectorAll(`input[disabled=disabled]`)]
   .filter(i =&gt; i.value.includes('repo') || i.value === 'security_events')
   .forEach(i =&gt; i.checked = false)
</code></pre>
<p>Not that I recommend doing that :cowboy_hat_face:</p>

**Comment by jcansdale** [comment]
<p><a class="mention" href="/ziegfried">@ziegfried</a>,</p>
<p>I admire your tenacity! :grinning_face_with_smiling_eyes:</p>
<p>A more low tech workaround is to navigate to the following url:<br>
<code>https://github.com/settings/tokens/new?scopes=write:packages</code></p>
<p>Hopefully this issue will be fixed soon!</p>

**Comment by samuel-weber** [comment]
<p>Thanks <a class="mention" href="/jcansdale">@jcansdale</a>! That worked! Just added delete:packages to the scopes (as mentioned in the docs):</p>
<p><a href="https://github.com/settings/tokens/new?scopes=write:packages,delete:packages" rel="noopener nofollow ugc">https://github.com/settings/tokens/new?scopes=write:packages,delete:packages</a></p>

**Comment by Bodundeft** [comment]
<p>I am new with git,<br>
I<br>
While I was trying to clone a repository, I entered this error message:<br>
fatal: Authentication failed for ‘<a href="https://github.com/Bodundeft/alx-pre_course.git/" rel="noopener nofollow ugc">https://github.com/Bodundeft/alx-pre_course.git/</a>’<br>
root@5eea4a03ae82:/# git clone<br>
Cloning into ‘alx-pre_course’…<br>
Password failed<br>
remote: Support for password authentication was removed on August 13, 2021. Please use a personal access token instead.<br>
remote: Please see <a href="https://github.blog/2020-12-15-token-authentication-requirements-for-git-operations/" class="inline-onebox" rel="noopener nofollow ugc">Token authentication requirements for Git operations | The GitHub Blog</a> for more information.<br>
fatal: Authentication failed for<br>
root@5eea4a03ae82:/# git clone</p>
<p>Pleased, I need help to get this sorted out.</p>
<p>Thanks in advance!!</p>
